bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/handler/HomeAssistantThingHandler.java

```diff
@@ -120,7 +124,7 @@ public HomeAssistantThingHandler(Thing thing, MqttChannelTypeProvider channelTyp
                 this.transformationServiceProvider);
     }
 
-    @SuppressWarnings({ "null", "unused" })
+    @SuppressWarnings({ "null" })
```

Comment:
Looks like this annotation is unnecessary and can be removed.

Reply:
If I remove it, we see warning about unnecessary null check in the method

```java
if(config.topics == null ....
```
Indeed, `HandlerConfiguration.topics` is declared as `@NonNull`. Having said that, I am not sure we can trust null annotations here since the config is parsed by `org.openhab.core.config.core.ConfigParser.configurationAs` method with no consideration of null annotations (if that is even possible).

I see this similar to how GSON deserialization, it is wrong to label fields as `@NonNull` since GSON can still slap `null` there when deserializing.

WDYT?

Reply:
This null check can also be removed, forgot to add the comment about it.
As I can see from sources configurationAs cannot set null.

Reply:
ok

Reply:
Actually: @antroids  is it not possible that value is not simply set in `configurationAs`?

This is quite clear case from the implementation -- it is logged with TRACE level. There also seems to (TODO?) comment `// Consider RequiredField annotations`

https://github.com/openhab/openhab-core/blob/2bceba69549c6017485e5089c64d6131889aa76a/bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/ConfigParser.java#L98-L102
![image](https://user-images.githubusercontent.com/315866/200885576-daad10df-69dc-41e6-91c0-df81da3aa555.png)

Is there some other validation in core to protect against incomplete configs?

Reply:
Appreciate that, included in commit https://github.com/openhab/openhab-addons/pull/13518/commits/77c59e5ed11f768e46b861af7f2e19288e081137

Attribution with `Co-Authored-By` commit message

Reply:
The default value will be used, empty list in this case.

Reply:
True! Thanks. https://github.com/openhab/openhab-addons/pull/13518/commits/b08e6a54364f26b7f5fc19b64a61e1a27960464a